tests/main/interfaces-mount-control-nfs/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: Test that with mount-control NFS shares can be mounted
+
+details: |
+    Verify that NFS mounts can be performed with mount-control
+
+# limit to systems where we know NFS works without much hassle
+systems:
+  - ubuntu-24.04-*
+
+prepare: |
+    # Mount should not leak
+    tests.cleanup defer NOMATCH 'nfs-share' /proc/self/mountinfo
+
+    # Create directory which the test will share
+    if [ ! -d /var/nfs-share ]; then
+      mkdir /var/nfs-share
+      tests.cleanup defer rm -r /var/nfs-share
+    fi
+    echo 'hello from NFS share' > /var/nfs-share/hello
+
+    # Install a package with additional kernel modules
+    if ! tests.pkgs install "linux-modules-extra-$(uname -r)"; then
+      echo "SKIP: Kernel version and extras module mismatch"
+      exit 1
+    fi
+
+    # Install nfs with some precautions to undo the side-effects if we are
+    # really installing it and it was not pre-installed. If /proc/fs/nfsd
+    # is not initially mounted then ask the test to unmount it later
+    # without checking if it is mounted (hence okfail wrapper).
+    if not mountinfo.query /proc/fs/nfsd .fs_type=nfsd; then
+      tests.cleanup defer okfail umount /proc/fs/nfsd
+    fi
+    # If /var/lib/nfs/rpc_pipefs is not initially mounted then ask the test
+    # to unmount it later.
+    if not mountinfo.query /var/lib/nfs/rpc_pipefs .fs_type=rpc_pipefs; then
+        tests.cleanup defer okfail umount /proc/nfs/rpc_pipefs
+    fi
+
+    tests.pkgs install nfs-kernel-server
+
+    # Export /var/home/test-remote over NFS.
+    mkdir -p /etc/exports.d/
+    echo '/var/nfs-share localhost(rw,no_subtree_check)' > /etc/exports.d/test.exports
+    retry exportfs -r
+
+    # Later on remove the exports file and reload exported filesystems.
+    tests.cleanup defer exportfs -r
+    tests.cleanup defer rm -f /etc/exports.d/test.exports
```

Comment:
This should move up immediately after the file is created in case something fails in between.